File: week_1/2309_.py
arr = []
ans = []

def dfs(depth, idx):
    if depth == 7:
        _sum = sum(ans)
        if _sum == 100:
            ans.sort()
            for i in ans:
                print(i)
            exit(0)
        else:
            return
    for i in range(depth, len(arr)):
        ans.append(arr[i])
        dfs(depth + 1, i + 1)
        ans.pop()

for i in range(9):
    temp = int(input())
    arr.append(temp)

dfs(0, 0)

File: week_1/exhaustive_search/2309.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> arr;
vector<int> ans;

void dfs(int depth, int idx)
{
    // 종료 조건
    if (depth == 7)
    {
        int sum = 0;
        for (auto& i : ans)
        {
            sum += i;
        }
        if (sum == 100)
        {
            sort(ans.begin(), ans.end());
            for (auto& i : ans)
            {
                cout << i << endl;
            }
            exit(0);
        }
        else
            return;
    }

    for (int i = depth; i < arr.size(); i++)
    {
        ans.push_back(arr[i]);
        dfs(depth + 1, i + 1);
        ans.pop_back();
    }
}


int main()
{
    // input
    for (size_t i = 0; i < 9; i++)
    {
        int temp;
        cin >> temp;
        arr.push_back(temp);
    }

    dfs(0, 0);

    return 0;
}